pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -688,6 +688,13 @@ func buildSidecarOutboundTCPListenerOptsForPortOrUDS(listenerMapKey *string,
 	var destinationCIDR string
 	if len(listenerOpts.bind) == 0 {
 		svcListenAddress := listenerOpts.service.GetAddressForProxy(listenerOpts.proxy)
+		// if listenerOpts.service is not the common k8s svc and headless svc, it may be a svc converted from service entry, etc.
+		// It's hard to detect IP family for this kind of service, such se with only host name, so follows the current proxy's
+		if listenerOpts.service.ClusterVIPs.IsEmpty() && svcListenAddress == constants.UnspecifiedIP {
```

Comment:
why not detect from ServiceRegistry?

Reply:
Hi @ramaraochavali ,  user may define ServiceRegistry with only hostname and without address, so sometimes we can not detect it from ServiceRegistry side.

Reply:
Thanks @ramaraochavali ! changed it